egs/wsj/s5/steps/segmentation/detect_speech_activity.sh

```diff
@@ -0,0 +1,235 @@
+test_data_dir=data/${data_id}${feat_affix}_hires
+
+if $convert_data_dir_to_whole; then
+  if [ $stage -le 0 ]; then
+    rm -r ${test_data_dir} || true
+    utils/data/convert_data_dir_to_whole.sh $src_data_dir ${test_data_dir}
+  fi
+else
+  if [ $stage -le 0 ]; then
+    rm -r ${test_data_dir} || true
+    utils/copy_data_dir.sh $src_data_dir $test_data_dir
+  fi
+fi
+
+###############################################################################
+## Extract input features 
+###############################################################################
+
+if [ $stage -le 1 ]; then
+  utils/fix_data_dir.sh $test_data_dir
+  steps/make_mfcc.sh --mfcc-config $mfcc_config --nj $nj --cmd "$cmd" \
+    ${test_data_dir} exp/make_hires/${data_id}${feat_affix} $mfcc_dir
+  steps/compute_cmvn_stats.sh ${test_data_dir} exp/make_hires/${data_id}${feat_affix} $mfcc_dir
+  utils/fix_data_dir.sh ${test_data_dir}
+fi
+
+###############################################################################
+## Forward pass through the network network and dump the log-likelihoods.
+###############################################################################
+
+frame_subsampling_factor=1
+if [ -f $sad_nnet_dir/frame_subsampling_factor ]; then
+  frame_subsampling_factor=$(cat $sad_nnet_dir/frame_subsampling_factor)
+fi
+
+mkdir -p $dir
+if [ $stage -le 4 ]; then
+  if [ "$(readlink -f $sad_nnet_dir)" != "$(readlink -f $dir)" ]; then
+    cp $sad_nnet_dir/cmvn_opts $dir || exit 1
+  fi
+
+  ########################################################################
+  ## Initialize neural network for decoding using the output $output_name
+  ########################################################################
+
+  if [ ! -z "$output_name" ] && [ "$output_name" != output ]; then
+    $cmd $dir/log/get_nnet_${output_name}.log \
+      nnet3-copy --edits="rename-node old-name=$output_name new-name=output" \
+      $sad_nnet_dir/$iter.raw $dir/${iter}_${output_name}.raw || exit 1
+    iter=${iter}_${output_name}
+  else 
+    cp $sad_nnet_dir/$iter.raw $dir/
+  fi
+
+  steps/nnet3/compute_output.sh --nj $nj --cmd "$cmd" \
+    --iter ${iter} \
+    --extra-left-context $extra_left_context \
+    --extra-right-context $extra_right_context \
+    --extra-left-context-initial $extra_left_context_initial \
+    --extra-right-context-final $extra_right_context_final \
+    --frames-per-chunk $frames_per_chunk --apply-exp true \
+    --frame-subsampling-factor $frame_subsampling_factor \
+    ${test_data_dir} $dir $sad_dir || exit 1
+fi
+
+###############################################################################
+## Prepare HCLG graph for decoding
+###############################################################################
+
+frame_shift=$(utils/data/get_frame_shift.sh $test_data_dir)
+
+graph_dir=${dir}/graph_${output_name}
+if [ $stage -le 5 ]; then
+  mkdir -p $graph_dir
+
+  # 1 for silence and 2 for speech
+  cat <<EOF > $graph_dir/words.txt
+1 1
+2 2
+EOF
```

Comment:
Also, adding `0 <eps>` would not hurt, even though there are no epsilons in the graph. This would help if something is done with the results that would introduce epsilons; openfst tools get very angry at you if they do not find a symbol. I am often dumping intermediate FSTs and sometimes heavily transform them for analysis and debugging, and have been stumbling over exactly the issue: missing the symbol for 0.